### PR TITLE
GattServer example

### PR DESCRIPTION
**Pull request by pan-**
This example exposes a service that model a clock. The clock second,
minute and hour can be accessed in separate characteristics.

A client can modify the value of the clock component as well as subscribed
to be notified of changes of its values.

### CODE REVIEW

**Comment by pan-** [comment]
@paul-szczepanek-arm Could you review at the example, readme is missing and I'm sure it can be improved.

File: BLE_GattServer/main.cpp [hunk 1]
```diff
@@ -0,0 +1,668 @@
+#include <stdio.h>
+
+#include "platform/Callback.h"
+#include "events/Eventqueue.h"
+#include "platform/NonCopyable.h"
+
+#include "ble/BLE.h"
```
Comment (paul-szczepanek-arm) [inline BLE_GattServer/main.cpp]: why are the individual headers included below if ble already includes them?

File: BLE_GattServer/main.cpp [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+#include "ble/GapAdvertisingParams.h"
+#include "ble/GapAdvertisingData.h"
+#include "ble/GattServer.h"
+
+using mbed::callback;
+
+/**
+ * A Clock service that demonstrate the GattServer features.
+ *
+ * The clock service host three characteristics that respectively model the
```
Comment (paul-szczepanek-arm) [inline BLE_GattServer/main.cpp]: language nitpick: the word respectively would only work if you listed the three characteristics in question (and would have to come after the word "model")

File: BLE_GattServer/main.cpp [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+ * current hour, minute and second. The value of the second characteristic is
+ * incremented automatically by the system.
+ *
+ * When the value of the second characteristic reach 60, it is reset to 0 and
```
Comment (paul-szczepanek-arm) [inline BLE_GattServer/main.cpp]: I think the explanation is redundant. Simply noting that the characteristics model a clock and are automatically incremented is enough.

File: BLE_GattServer/main.cpp (hunk 2 above)
Comment (pan-) [reply BLE_GattServer/main.cpp]: Cheers, I will amend the comment.

File: BLE_GattServer/main.cpp [hunk 4]
```diff
@@ -0,0 +1,668 @@
...
+            printf("Error %u during gap.setAdvertisingPayload\r\n", error);
+            return;
+        }
+
+        gap.setAdvertisingParams(make_advertising_params());
+
+        gap.onConnection(this, &BLEProcess::when_connection);
+        gap.onDisconnection(this, &BLEProcess::when_disconnection);
+
+        start_advertising();
```
Comment (paul-szczepanek-arm) [inline BLE_GattServer/main.cpp]: why no error check?

File: BLE_GattServer/main.cpp [hunk 5]
```diff
@@ -0,0 +1,668 @@
...
+    // demo service
+    GattService _clock_service;
+
+    GattServer* _server;
+    events::EventQueue *_event_queue;
+};
+
+
+/**
+ * Handle initialization adn shutdown of the BLE Instance.
```
Comment (paul-szczepanek-arm) [inline BLE_GattServer/main.cpp]: "adn" typo

File: BLE_GattServer/main.cpp [hunk 6]
```diff
@@ -0,0 +1,668 @@
...
+        printf("\t\ttextual description handle %u\r\n", _minute_char_description.getHandle());
+        printf("\tsecond characteristic value handle %u\r\n", _second_char.getValueHandle());
+        printf("\t\ttextual description handle %u\r\n", _second_char_description.getHandle());
+
+        _event_queue->call_every(1000 /* ms */, callback(this, &Self::increment_second));
+    }
+
+private:
+
+//
```
Comment (paul-szczepanek-arm) [inline BLE_GattServer/main.cpp]: why not doxygen?

File: BLE_GattServer/main.cpp [hunk 7]
```diff
@@ -0,0 +1,668 @@
...
+    {
+        printf("update disabled on handle %d\r\n", handle);
+    }
+
+    void when_confirmation_received(GattAttribute::Handle_t handle)
+    {
+        printf("confirmation received on handle %d\r\n", handle);
+    }
+
+//
```
Comment (paul-szczepanek-arm) [inline BLE_GattServer/main.cpp]: these three seem repetitive, maybe characteristic should be a class to avoid all functions having to be in triplicate?

**Comment by pan-** [comment]
@paul-szczepanek-arm Could you review my changes ?

**Pull request merged**